weepickle-implicits/src/main/scala-3/com/rallyhealth/weepickle/v1/implicits/CaseClassTo.scala

```diff
@@ -0,0 +1,197 @@
+package com.rallyhealth.weepickle.v1.implicits
+
+import compiletime.{summonInline}
+import deriving.Mirror
+import scala.util.control.NonFatal
+import com.rallyhealth.weepickle.v1.core.{Abort, Annotator, ObjVisitor, NoOpVisitor, Types, Visitor}
+
+trait CaseClassToPiece extends MacrosCommon :
+  this: Types with Annotator =>
+
+  // must duplicate ToString definition here to avoid circular reference with DefaultTos
+  implicit val stringVisitor: To[String] = new SimpleTo[String] {
+    override def expectedMsg = "expected string"
+    override def visitString(cs: CharSequence): String = cs.toString
+  }
+
+  class CaseClassTo[T](
```

Comment:
```suggestion
  private class CaseClassTo[T](
```

Reply:
I can make the class private, but the constructor needs to remain public (which is the default), otherwise
```
[error] 94 |      val fromCaseClass = CaseClassFrom[T](
[error]    |                          ^^^^^^^^^^^^^
[error]    |Implementation restriction: cannot use private constructors in inlineinline methods
```

Reply:
Will we be able to change the class later without breaking bin compat?

Reply:
idk, and update: even making it private causes issues (not sure why I didn't see it earlier -- maybe I forgot to `+clean`):
```
[error] -- Error: /Users/russell-remple/IdeaProjects/weePickle/bench/src/main/scala/bench/ScalaVersionBench.scala:70:36 
[error] 70 |    implicit val rw: FromTo[ADT0] = macroFromTo
[error]    |                                    ^^^^^^^^^^^
[error]    |undefined: new CaseClassToPiece_this.CaseClassTo[bench.ScalaVersionBench.ADT0] # -1: TermRef(AppliedType(TypeRef(TermRef(NoPrefix,val CaseClassToPiece_this),CaseClassTo),List(TypeRef(ThisType(TypeRef(ThisType(TypeRef(NoPrefix,module class bench)),module class ScalaVersionBench$)),class ADT0))),<init>) at inlining
[error]    | This location contains code that was inlined from CaseClassTo.scala:110
```

Reply:
Would it be possible to make it an anonymous class?

Reply:
done